src/protocols/ssh/ssh.c

```diff
@@ -130,26 +130,36 @@ static guac_common_ssh_user* guac_ssh_get_user(guac_client* client) {
 
     } /* end if key given */
 
-    /* Otherwise, use password */
-    else {
-
-        /* Get password if not provided */
-        if (settings->password == NULL)
-            settings->password = guac_terminal_prompt(ssh_client->term,
-                    "Password: ", false);
-
-        /* Set provided password */
-        guac_common_ssh_user_set_password(user, settings->password);
-
-    }
-
     /* Clear screen of any prompts */
     guac_terminal_printf(ssh_client->term, "\x1B[H\x1B[J");
 
     return user;
 
 }
 
+/**
+ * A call-back function used to gather additional credentials from a client
```

Comment:
This looks like documentation intended for the `guac_ssh_credential_handler` type. It makes sense from a generic perspective, but the information provided is only sufficient for a developer trying to implement such a callback, not for a developer potentially wanting to use this existing implementation.

It should be clear from the documentation of the function why this particular implementation is appropriate from the perspective of a developer that may want to use this. It prompts the user for credentials using the terminal emulator associated with the SSH connection via `guac_client`? Probably good to note.

Reply:
Reworded this a bit and added the notes as suggested.